dao/elasticsearch/controlplanedao_test.go

```diff
@@ -145,6 +145,94 @@ func (dt *DaoTest) TearDownSuite(c *C) {
 	dt.FacadeTest.TearDownSuite(c)
 }
 
+func (dt *DaoTest) TestDao_ValidateEndpoints(t *C) {
+	var id string
+
+	// test 1: add tenant with dup ep
+	svc := service.Service{
+		ID:           "test_tenant",
+		Name:         "test_tenant",
+		PoolID:       "default",
+		Launch:       "auto",
+		DeploymentID: "deployment_id",
+		Endpoints: []service.ServiceEndpoint{
+			{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+			{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+		},
+	}
+	err := dt.Dao.AddService(svc, &id)
+	t.Assert(err, NotNil)
+
+	// test 2: success
+	svc.Endpoints = []service.ServiceEndpoint{
+		{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+	}
+	err = dt.Dao.AddService(svc, &id)
+	t.Assert(err, IsNil)
+
+	// test 3: update service with dup ep
+	svc.Endpoints = []service.ServiceEndpoint{
+		{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+		{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+	}
+	err = dt.Dao.UpdateService(svc, &unused)
+	t.Assert(err, NotNil)
+
+	// test 4: add child service with dup ep
+	svc2 := service.Service{
+		ID:              "test_service_1",
+		Name:            "test_service_1",
+		ParentServiceID: svc.ID,
+		PoolID:          svc.PoolID,
+		Launch:          svc.Launch,
+		DeploymentID:    svc.DeploymentID,
+		Endpoints: []service.ServiceEndpoint{
+			{EndpointDefinition: servicedefinition.EndpointDefinition{Name: "test_ep_1", Application: "test_ep_1", Purpose: "export"}},
+		},
+	}
+	err = dt.Dao.AddService(svc2, &id)
+	t.Assert(err, NotNil)
```

Comment:
Should we test the value of `err` to verify that the test case is failing the right reason?